src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -169,6 +172,42 @@ def _autoclass_converter(
     return tuple(it if isinstance(it, str) else it.__qualname__ for it in value)
 
 
+def _extract_commit_hash(
+    resolved_dir: str, regex_commit_hash: t.Pattern[str]
+) -> str | None:
+    """
+    Extracts the commit hash from a resolved filename toward a cache file.
+    modified from https://github.com/huggingface/transformers/blob/0b7b4429c78de68acaf72224eb6dae43616d820c/src/transformers/utils/hub.py#L219
+    """
+
+    resolved_dir = str(Path(resolved_dir).as_posix()) + "/"
+    search = re.search(r"snapshots/([^/]+)/", resolved_dir)
+
+    if search is None:
+        return None
+
+    commit_hash = search.groups()[0]
+    return commit_hash if regex_commit_hash.match(commit_hash) else None
+
+
+def _try_import_huggingface_hub():
+    try:
+        import huggingface_hub  # noqa: F401
+    except ImportError:  # pragma: no cover
+        raise MissingDependencyException(
+            "'huggingface_hub' is required in order to download pretrained transformer models, install with 'pip install huggingface-hub'. For more information, refer to https://huggingface.co/docs/huggingface_hub/quick-start",
+        )
+
+
+def _try_import_accelerate():
+    try:
+        import accelerate  # noqa: F401
+    except ImportError:  # pragma: no cover
+        raise MissingDependencyException(
+            "'accelerate' is required in order to download pretrained transformer models, install with 'pip install accelerate'.",
+        )
+
+
```

Comment:
lets refactor this under `bentoml/_internal/frameworks/utils/transformers.py` to use for both diffusers and transformers.

Reply:
And you can make it `is_accelerate_available` and `is_huggingface_hub_available`?

Reply:
agree